backend/src/packages/achievements/achievement.repository.ts

```diff
@@ -29,14 +40,88 @@ class AchievementRepository implements IRepository {
     return achievements.map((it) => AchievementEntity.initialize(it));
   }
 
+  public async findAllUserAchievements(
+    userId: number,
+  ): Promise<UserAchievement[]> {
+    const achievements = await this.userAchievementModel
+      .query()
+      .where({ userId })
+      .execute();
+
+    return achievements.map((userAchievement) => ({
+      id: userAchievement.id,
+      userId: userAchievement.userId,
+      achievementId: userAchievement.achievementId,
+    }));
+  }
+
+  public async findBy(
+    filters: Partial<Achievement> = {},
```

Comment:
As they are only ones used
```suggestion
    filters: Pick<Achievement, 'breakpoint' | 'referenceTable'>,
```